### ISSUE TITLE
Edit: Fabric/Forge 1.16

### ISSUE THREAD

**Issue by UltraBlackLinux** [kurrycat2004/MpkMod#25]
Hey there,
what about fully open-sourcing this mod?
Provides safety to not have a tokean stealer installed ;)
Thanks!

(Any chance of a fabric version?)

**Comment by kurrycat2004** [comment]
tl;dr: i cant make it open source because of the macro feature and the possibility to enable it in mp. a fabric version would be possible but i would try to make a single jar support forge and fabric which is a big project i dont want to start rn

it was open source in the early versions but when i added macros i had to make it closed source because of the possibility of people enabling macros for multiplayer.
By making it closed source and obfuscating the jar i didnt really make it impossible to enable macros in mp but u made it much harder.

i dont really know fabric and i never looked into it but i could try to make a fabric version for it. but the thing is that i dont like having multiple completely different versions, because i know from the 1.12 and 1.8 jar how hard it is to always update both versions which is also the reason why i decided to make a single jar support 1.8 and 1.12 for v1.2.5 and higher.
If i would find a way to make a jar support forge and fabric i would probably implement that if i have time but rn there are many other things i want to add to the mod. especially the config remake because it makes the mod laggy af

**Comment by UltraBlackLinux** [comment]
@kurrycat2004 
If you update it to 1.16/1.14, patchwork can do the work for you - I guess that's the easiest way.

**Comment by kurrycat2004** [comment]
welp 1.14/1.16 is completely different from 1.8 or 1.12. There already are some things in 1.8 that got removed in 1.12 (cancelable right click event im looking at you) and where i need to find a different solution. I guess if i really have the time i will do it but it wont be fun

**Comment by Sargeanthost** [comment]
I guess would you could do is make a branch without a macro capability. I was actually thinking about making an issue about this topic today, lol.

This wouldn't solve UltraBlackLinux's (non)problem of malicious code, but it would be nice to see how you do things in the mod, especially for beginners(cough cough me).

**Comment by kurrycat2004** [comment]
I also thought about making a version without macro that is open source but again i would have to update 2 versions every time. Something like just adding the macro java file to .gitignore could work but it would still show the code where it prevents the use for mp so people could change it more easily

also some parts of the code are really messy and others are just garbage (config system lol)

**Comment by UltraBlackLinux** [comment]
> I also thought about making a version without macro that is open source but again i would have to update 2 versions every time. Something like just adding the macro java file to .gitignore could work but it would still show the code where it prevents the use for mp so people could change it more easily
> 
> also some parts of the code are really messy and others are just garbage (config system lol)

The best way to do it, is actually making it annoying to find the variable for the macro, make references, encrypt the value with useless stuff, and more, just make it very annoying to find it :)

**Comment by kurrycat2004** [comment]
> > I also thought about making a version without macro that is open source but again i would have to update 2 versions every time. Something like just adding the macro java file to .gitignore could work but it would still show the code where it prevents the use for mp so people could change it more easily
> > also some parts of the code are really messy and others are just garbage (config system lol)
> 
> The best way to do it, is actually making it annoying to find the variable for the macro, make references, encrypt the value with useless stuff, and more, just make it very annoying to find it :)

but again its much easier for people to find and remove the code if its open source than first having to decompile the jar and then just getting obfuscated code even if i would add useless stuff that makes it annoying to find

**Comment by UltraBlackLinux** [comment]
> > > I also thought about making a version without macro that is open source but again i would have to update 2 versions every time. Something like just adding the macro java file to .gitignore could work but it would still show the code where it prevents the use for mp so people could change it more easily
> > > also some parts of the code are really messy and others are just garbage (config system lol)
> > 
> > 
> > The best way to do it, is actually making it annoying to find the variable for the macro, make references, encrypt the value with useless stuff, and more, just make it very annoying to find it :)
> 
> but again its much easier for people to find and remove the code if its open source than first having to decompile the jar and then just getting obfuscated code even if i would add useless stuff that makes it annoying to find

If users really want to use this mod in multiplayer, they are just gonna use a decompiler and remove the faulty code xD

**Comment by kurrycat2004** [comment]
its obfuscated. having it open source would remove the extra step of having to decompile it and you would have the unobfuscated code which makes it much easier to find and remove the code

**Comment by RedthMC** [comment]
u could just separate them into 2 mods
and having only the macro mod obfuscated

**Comment by kurrycat2004** [comment]
yeah i also thought about doing that but currently the thing that makes it hard to remove the code that disables mp isnt the obfuscation itself but the amount of code obfuscated code one would have to search through

**Comment by Sargeanthost** [comment]
> yeah i also thought about doing that but currently the thing that makes it hard to remove the code that disables mp isnt the obfuscation itself but the amount of code obfuscated code one would have to search through

I'm pretty sure you can ctrl+f and delete all instances

**Comment by kurrycat2004** [comment]
> > yeah i also thought about doing that but currently the thing that makes it hard to remove the code that disables mp isnt the obfuscation itself but the amount of code obfuscated code one would have to search through
> 
> I'm pretty sure you can ctrl+f and delete all instances 

what? its not only one method or variable that you have to remove to enable macros in mp

**Comment by UltraBlackLinux** [comment]
> > > yeah i also thought about doing that but currently the thing that makes it hard to remove the code that disables mp isnt the obfuscation itself but the amount of code obfuscated code one would have to search through
> > 
> > 
> > I'm pretty sure you can ctrl+f and delete all instances
> 
> what? its not only one method or variable that you have to remove to enable macros in mp

Then you should propably make it like this. It makes the code cleaner and makes you able to do such things like this.

**Comment by Sargeanthost** [comment]
> > > > yeah i also thought about doing that but currently the thing that makes it hard to remove the code that disables mp isnt the obfuscation itself but the amount of code obfuscated code one would have to search through
> > > 
> > > 
> > > I'm pretty sure you can ctrl+f and delete all instances
> > 
> > 
> > what? its not only one method or variable that you have to remove to enable macros in mp
> 
> Then you should propably make it like this. It makes the code cleaner and makes you able to do such things like this.

What?? I don't think you understand...

**Comment by kurrycat2004** [comment]
> > > > yeah i also thought about doing that but currently the thing that makes it hard to remove the code that disables mp isnt the obfuscation itself but the amount of code obfuscated code one would have to search through
> > > 
> > > 
> > > I'm pretty sure you can ctrl+f and delete all instances
> > 
> > what? its not only one method or variable that you have to remove to enable macros in mp
> 
> Then you should propably make it like this. It makes the code cleaner and makes you able to do such things like this.

i dont think you understand what i mean. its a good thing that there is a large amount of code because that increases the time one would have to search through to remove the mp limitation for macros. if i would just have the macros in a single mod it would be much much easier to find the parts of the code that disable macros for mp

**Comment by UltraBlackLinux** [comment]
Hey @kurrycat2004 I just tried to randomly decompile your jar, and I instandly found what you need to change, without even deobfuscating it, to be able to run it on a server:

![Screenshot_0211_170245](https://user-images.githubusercontent.com/62404294/107662899-0c723f80-6c8b-11eb-9dad-7a50152a79d1.png)
lol.

**Comment by Sargeanthost** [comment]
> Hey @kurrycat2004 I just tried to randomly decompile your jar, and I instandly found what you need to change, without even deobfuscating it, to be able to run it on a server:
> 
> ![Screenshot_0211_170245](https://user-images.githubusercontent.com/62404294/107662899-0c723f80-6c8b-11eb-9dad-7a50152a79d1.png)
> lol.

they're are a few more you need to change

**Comment by kurrycat2004** [comment]
bruh thats the ping label xDDD
it just shows the text "SinglePlayer" if you are in a sp world

**Comment by kurrycat2004** [comment]
and yes i added mutltiple things that prevent sp

**Comment by UltraBlackLinux** [comment]
> bruh thats the ping label xDDD
> it just shows the text "SinglePlayer" if you are in a sp world

Whatever XD

**Comment by Sargeanthost** [comment]
> > bruh thats the ping label xDDD
> > it just shows the text "SinglePlayer" if you are in a sp world
> 
> Whatever XD

Theres a github repo of mcp funcs, so if you want to remove that you can lol
[here](https://github.com/KevyPorter/Minecraft-Forge-Utils/blob/master/fields.csv)

**Comment by RedthMC** [comment]
![image](https://user-images.githubusercontent.com/59755409/107722100-2eb29e80-6d19-11eb-9b1c-c80e667bdf43.png)
found in just 1 min lol

**Comment by Sargeanthost** [comment]
> ![image](https://user-images.githubusercontent.com/59755409/107722100-2eb29e80-6d19-11eb-9b1c-c80e667bdf43.png)
> found in just 1 min lol

now find all the others :)

**Comment by Red1206** [comment]
> > ![image](https://user-images.githubusercontent.com/59755409/107722100-2eb29e80-6d19-11eb-9b1c-c80e667bdf43.png)
> > found in just 1 min lol
> 
> now find all the others :)

wdym all the others

**Comment by RedthMC** [comment]
wdym all the others

**Comment by kurrycat2004** [comment]
Please just dont try to remove the code parts that prevent users from using macros in mp. I added them for a reason

**Comment by UltraBlackLinux** [comment]
> Please just dont try to remove the code parts that prevent users from using macros in mp. I added them for a reason

let me close this, to prevent more attention on finding the important code pieces ;)
